internal/controller/datadogagent/override/global.go

```diff
@@ -435,3 +448,171 @@ func applyGlobalSettings(logger logr.Logger, manager feature.PodTemplateManagers
+		apiKeySecretName = global.Credentials.APISecret.SecretName
+		apiKeySecretKey = global.Credentials.APISecret.KeyName
+	}
+	if isValidSecretConfig(global.Credentials.AppSecret) {
+		appKeySecretName = global.Credentials.AppSecret.SecretName
+		appKeySecretKey = global.Credentials.AppSecret.KeyName
+	}
+
+	// Add secret env vars to pod
+	apiKeyEnvVar := common.BuildEnvVarFromSource(constants.DDAPIKey, common.BuildEnvVarFromSecret(apiKeySecretName, apiKeySecretKey))
+	podTemplateManager.EnvVar().AddEnvVar(apiKeyEnvVar)
+
+	if appKeySecretName != "" {
+		appKeyEnvVar := common.BuildEnvVarFromSource(constants.DDAppKey, common.BuildEnvVarFromSecret(appKeySecretName, appKeySecretKey))
+		podTemplateManager.EnvVar().AddEnvVar(appKeyEnvVar)
+	}
+}
+
+// handleDCAToken will be split between dependency and pod changes when global is refactored
+func handleDCAToken(logger logr.Logger, dda *v2alpha1.DatadogAgent, resourcesManager feature.ResourceManagers, podTemplateManager feature.PodTemplateManagers) error {
+	if err := dcaTokenDependencies(logger, dda, resourcesManager); err != nil {
+		return err
+	}
+	dcaTokenResource(logger, dda, resourcesManager, podTemplateManager)
+	return nil
+}
+
+func dcaTokenDependencies(logger logr.Logger, dda *v2alpha1.DatadogAgent, resourcesManager feature.ResourceManagers) error {
+	global := dda.Spec.Global
+	var token string
+
+	// Prioritize existing secret
+	if isValidSecretConfig(global.ClusterAgentTokenSecret) {
+		return nil
+	}
+
+	// User specifies token
+	var key string
+	var hash string
+	var err error
+	if global.ClusterAgentToken != nil && *global.ClusterAgentToken != "" {
+		token = *global.ClusterAgentToken
+		// Generate hash
+		key = getDCATokenChecksumAnnotationKey()
+		hash, err = comparison.GenerateMD5ForSpec(map[string]string{common.DefaultTokenKey: token})
+		if err != nil {
+			logger.Error(err, "couldn't generate hash for Cluster Agent token hash")
+		} else {
+			logger.Info("built Cluster Agent token hash", "hash", hash)
+			// logger.V(2).Info("built Cluster Agent token hash", "hash", hash)
+		}
+	} else if dda.Status.ClusterAgent == nil || dda.Status.ClusterAgent.GeneratedToken == "" { // no token specified
+		token = apiutils.GenerateRandomString(32)
+	} else {
+		token = dda.Status.ClusterAgent.GeneratedToken // token already generated
+	}
+
+	// Create secret
+	secretName := secrets.GetDefaultDCATokenSecretName(dda)
+	if err := resourcesManager.SecretManager().AddSecret(dda.Namespace, secretName, common.DefaultTokenKey, token); err != nil {
+		return err
+	}
+
+	if key != "" && hash != "" {
+		// Add annotation to secret
+		if err := resourcesManager.SecretManager().AddAnnotations(logger, dda.Namespace, secretName, map[string]string{key: hash}); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func dcaTokenResource(logger logr.Logger, dda *v2alpha1.DatadogAgent, resourcesManager feature.ResourceManagers, podTemplateManager feature.PodTemplateManagers) {
```

Comment:
```suggestion
func dcaTokenResource(_ logr.Logger, dda *v2alpha1.DatadogAgent, resourcesManager feature.ResourceManagers, podTemplateManager feature.PodTemplateManagers) {
```
we do not use the logger (yet)